### PR TITLE
AXI_ADXCVR driver improvements

### PR DESCRIPTION
**Pull request by danmois**
Improvements of the AXI_ADXCVR driver:
- the addition of Doxygen comments
- fix error handling
- the removal of warnings generated during build
- addition of defines for AXI_ADXCVR specific settings
- improvement of the fmcdaq2 sample project using the driver

### CODE REVIEW

File: drivers/axi_core/jesd204/xilinx_transceiver.h [hunk 1]
```diff
@@ -156,76 +188,106 @@ struct xilinx_xcvr {
...
 
+/* Encoding */
 #define ENC_8B10B		810
 #define ENC_66B64B		6664
 
 /******************************************************************************/
 /************************ Functions Declarations ******************************/
 /******************************************************************************/
+
+/*! Configure the Clock Data Recovery circuit. */
```
Comment (amiclaus) [inline drivers/axi_core/jesd204/xilinx_transceiver.h:220]: nit: `/*!`  -> `/**`

File: drivers/axi_core/jesd204/axi_adxcvr.c [hunk 2]
```diff
@@ -501,16 +501,16 @@ int32_t adxcvr_init(struct adxcvr **ad_xcvr,
 
 	if (xcvr->lane_rate_khz && xcvr->ref_rate_khz) {
 		ret = adxcvr_clk_set_rate(xcvr, xcvr->lane_rate_khz, xcvr->ref_rate_khz);
-		if (ret)
-			goto err;//ToDo If it fails here, then you won't get its name in the caller, because following instr. not executed.
+		if (ret) {
```
Comment (amiclaus) [inline drivers/axi_core/jesd204/axi_adxcvr.c:508]: why adding brackets?

File: projects/fmcdaq2/src/app/fmcdaq2.c [hunk 3]
```diff
@@ -339,20 +339,18 @@ static int fmcdaq2_jesd_init(struct fmcdaq2_init_param *dev_init)
 	dev_init->ad9144_xcvr_param = (struct adxcvr_init) {
 		.name = "ad9144_xcvr",
 		.base = XPAR_AXI_AD9144_XCVR_BASEADDR,
-		.sys_clk_sel = 3,
+		.sys_clk_sel = ADXCVR_SYS_CLK_QPLL0,
```
Comment (amiclaus) [inline projects/fmcdaq2/src/app/fmcdaq2.c:342]: i think this commit can be squashed with the one adding the macro definitions

File: projects/fmcdaq2/src/app/fmcdaq2.c [hunk 4]
```diff
@@ -340,7 +340,7 @@ static int fmcdaq2_jesd_init(struct fmcdaq2_init_param *dev_init)
 		.name = "ad9144_xcvr",
 		.base = XPAR_AXI_AD9144_XCVR_BASEADDR,
 		.sys_clk_sel = ADXCVR_SYS_CLK_QPLL0,
-		.out_clk_sel = 4,
+		.out_clk_sel = ADXCVR_REFCLK_DIV2,
```
Comment (amiclaus) [inline projects/fmcdaq2/src/app/fmcdaq2.c:343]: same with this commit. it can be squashed with the one adding the macro definitions

**Review by amiclaus — COMMENTED** [review]
some small comments on my side. lgtm otherwise. please also mind the fact that a PR is preferred to that maximum 5-6 commits

File: drivers/axi_core/jesd204/xilinx_transceiver.h (hunk 1 above)
Comment (amiclaus) [reply drivers/axi_core/jesd204/xilinx_transceiver.h:220]: check entire code for this

File: drivers/axi_core/jesd204/axi_adxcvr.c (hunk 2 above)
Comment (danmois) [reply drivers/axi_core/jesd204/axi_adxcvr.c]: Thank you for the observation. The brackets remained there after the development phase. Removed.

**Comment by danmois** [comment]
V2
- squashed commits
- removed unused code
- updated function comments

**Review by amiclaus — APPROVED** [review]
lgtm

**Comment by danmois** [comment]
V3
- modifications of sample projects so that they use the newly defined macros for PLL selection and out clock selection

**Review by amiclaus — APPROVED** [review]

**Pull request merged**